### DISCUSSION TITLE
Release notes

### DISCUSSION THREAD

**Discussion by DinTri** [rfennell/AzurePipelines#1296] [General]
Hi Richard, 
I want to ask you if everything will be okay to generate release notes if in Azure DevOps I don't have Commits, Pushes and Branches but only  ChangeSets and Shelvesets.

Thank you.

**Comment by DinTri** [comment]
Hi Richard,
one more question. Can I generate release notes if I use Team Foundation Version Control.

**Reply by rfennell** [reply]
Yes, you, but the properties shown obviousy change between a commit and a changeset
